### PR TITLE
allow pixelfunc.get_interp_val to operate on multiple maps

### LINKED ISSUE #815: Vectorize hp.get_interp_val over multiple input maps
It would be really useful performance-wise to be able to call `hp.get_interp_val` with a sequence of maps and have the interpolations for all the maps, theta, and phi be computed in a vectorized fashion. It should be possible to do this manually via `hp.get_interp_weights` and some array manipulation, but it would be really nice if `hp.get_interp_val` supported it out of the box.

### PR DESCRIPTION
**Pull request by jvavrek**
This update allows `pixelfunc.get_interp_val` to operate on multiple maps in a vectorized fashion, which should be a significant performance improvement over re-doing multiple interpolations in a for loop.

As far as I can tell so far, I have preserved the existing interface and tests. I've also added a new doctest for the multi-map case.

Let me know if any edits are required!

Closes #815

### CODE REVIEW

**Comment by jvavrek** [comment]
Here's an MWE showing a ~40x speedup:
```py
import healpy as hp

def interpolate_multiple(maps, polar, azimuth):
    return [hp.get_interp_val(m, polar, azimuth) for m in maps]

def interpolate_vectorized(maps, polar, azimuth):
    return hp.get_interp_val(maps, polar, azimuth)

nreps = 1000
maps = np.asarray([np.arange(192)] * nreps)
azimuth = np.linspace(0, 2*np.pi, 50)
polar = np.linspace(0, np.pi, 50)

%timeit interpolate_multiple(maps, polar, azimuth)
# 33.5 ms ± 2.68 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

%timeit interpolate_vectorized(maps, polar, azimuth)
# 796 µs ± 18 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

np.array_equal(
    interpolate_multiple(maps, polar, azimuth),
    interpolate_vectorized(maps, polar, azimuth),
)
# True
```

**Comment by jvavrek** [comment]
@zonca it looks like you will have to approve the CI workflow here

**Comment by jvavrek** [comment]
@zonca will this be good to merge soon? I'd love to get it into production!

File: healpy/pixelfunc.py [hunk 1]
```diff
@@ -1365,7 +1373,7 @@ def get_interp_val(m, theta, phi, nest=False, lonlat=False):
     p = np.array(r[0:4])
     w = np.array(r[4:8])
     del r
-    return np.sum(m2[p] * w, 0)
+    return np.squeeze(np.sum(m[:, p] * w, axis=1))[()]
```
Comment (zonca) [inline healpy/pixelfunc.py:1376]: why are you using `[()]`?

Comment (zonca) [reply healpy/pixelfunc.py:1376]: ok, can you add a comment explaining that? it is obscure syntax

Comment (jvavrek) [reply healpy/pixelfunc.py:1377]: Sure, I've now added that. The CI will need to be approved to run again, it looks like.

**Comment by zonca** [comment]
Sorry I forgot. Can you add a line to the change log?

**Comment by jvavrek** [comment]
> Sorry I forgot. Can you add a line to the change log?

Sure, under "Unreleased"?

**Comment by zonca** [comment]
yes

**Review by zonca — APPROVED** [review]
Thank you very much for implementing this!

**Comment by jvavrek** [comment]
Sure, happy to contribute!

**Pull request merged**